stingray/io.py

```diff
@@ -653,3 +660,66 @@ def get_btis(gtis, start_time=None, stop_time=None):
 def gti_len(gti):
     """Return the total good time from a list of GTIs."""
     return np.sum([g[1] - g[0] for g in gti])
+
+def _save_pickle_object(object, filename):
+    pickle.dump(object, open(filename, "wb" ))
+
+def _retrieve_pickle_object(filename):
+    return pickle.load(open(filename, "rb" ) )
+
+def _save_hdf5_object(object, filename):
+    pass
+
+def _retrieve_hdf5_object(object, filename):
+    pass
+
+def _save_ascii_object(object, filename):
+    pass
+
+def _retrieve_ascii_object(object, filename):
+    pass
+
+def save_object(object, filename, format):
+    """
+    Pickle a class instance.
+
+    Parameters
+    ----------
+    object: a class instance
```

Comment:
I pretty much agree with all of @OrkoHunter's comments. I like having `io.read()` and `io.write()` as general input/output functions that we can then call from within classes. 

And yes, we shouldn't overload `object`.

As for the default value for `format`, based on what I know about astronomers and their preferences, I'd make it either `ascii` or `fits`, but @matteobachetti might disagree and say it should be `hdf5`. :)